pkg/operator/apis/monitoring/v1/http_types.go

```diff
@@ -101,11 +101,8 @@ func (s *SecretKeySelector) toPrometheusSecretRef(m PodMonitoringCRD, pool Prome
 	}
```

Comment:
It would be more clear to move the whole if statement here. We should be defaulting to whatever the PodMonitoring's namespace is _instead_ of the default namespace, e.g:

```go
ns := s.Namespace
if ns == "" {
	if m.IsNamespaceScoped() {
		ns = m.GetNamespace()
	} else {
		ns = metav1.NamespaceDefault
	}
}
```